src/systems/systems/src/indicating_recording/controls/keyboard.rs

```diff
@@ -0,0 +1,159 @@
+use crate::{
+    indicating_recording::controls::keyboard_cursor_control_unit::Button,
+    shared::{
+        arinc825::{Arinc825Word, LogicalCommunicationChannel},
+        can_bus::CanBus,
+        power_supply_relay::PowerSupplyRelay,
+        ElectricalBusType,
+    },
+    simulation::{
+        InitContext, Read, SimulationElement, SimulationElementVisitor, SimulatorReader,
+        VariableIdentifier,
+    },
+};
+
+pub struct Keyboard {
+    power_supply: PowerSupplyRelay,
+    keys: [Button; 58],
+    switch_kbd_id: VariableIdentifier,
+    switch_kbd_value: f64,
```

Comment:
see rust guidelines.md and readme.md ideally systems should not be aware they are running in a simulator and simvar state should be passed and vars not being read here i.e. the system is unaware that it is running in a simulator

Reply:
How should it look like?
I need to read somewhere the variables and don't know where it makes more sense in the end

Reply:
Resolved in development channel on discord